python/GafferSceneUI/ShaderQueryUI.py

```diff
@@ -0,0 +1,426 @@
+            if the parameter does not exist. The order of the child plugs 
+            correspond to the order of the query plugs.
+            """,
+
+            "plugValueWidget:type", "GafferUI.LayoutPlugValueWidget",
+
+            "layout:section", "Settings.Values",
+
+            "nodule:type", "GafferUI::CompoundNodule",
+
+            "noduleLayout:section", "right",
+            "noduleLayout:spacing", 0.2,
+            
+        ],
+
+        "values..." : [
+
+            "label", functools.partial( __getLabel, parentPlug = "values" ),
+            "noduleLayout:label", functools.partial( __getLabel, parentPlug = "values" ),
+
+            "noduleLayout:section", "right",
+
+        ],
+
+    }
+)
+
+
+##########################################################################
+# _ShaderQueryFooter
+##########################################################################
+
+class _ShaderQueryFooter( GafferUI.PlugValueWidget ) :
+
+    def __init__( self, plug ) :
+
+        row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal )
+
+        GafferUI.PlugValueWidget.__init__( self, row, plug )
+
+        with row :
+            
+            GafferUI.Spacer( imath.V2i( GafferUI.PlugWidget.labelWidth(), 1 ) )
+
+            GafferUI.MenuButton(
+                image = "plus.png",
+                hasFrame = False,
+                menu = GafferUI.Menu( Gaffer.WeakMethod( self.__menuDefinition ) )
+            )
+
+            GafferUI.Spacer( imath.V2i( 1 ), imath.V2i( 999999, 1 ), parenting = { "expand": True } )
+
+        plug.node().plugSetSignal().connect(
+            Gaffer.WeakMethod( self.__updateQueryMetadata ),
+            scoped = False
+        )
+
+    def _updateFromPlug( self ) :
+        
+        self.setEnabled( self._editable() )
+
+    def __menuDefinition( self ) :
+
+        result = IECore.MenuDefinition()
+
+        result.append(
+            "/From Location",
+            {
+                "subMenu" : Gaffer.WeakMethod( self.__addFromLocationMenuDefinition )
+            }
+        )
+
+        result.append( "/FromLocationDivider", { "divider" : True } )
+
+        for item in [
+            Gaffer.BoolPlug,
+            Gaffer.FloatPlug,
+            Gaffer.IntPlug,
+            "NumericDivider",
+            Gaffer.StringPlug,
+            "StringDivider",
+            Gaffer.V2iPlug,
+            Gaffer.V3iPlug,
+            Gaffer.V2fPlug,
+            Gaffer.V3fPlug,
+            "VectorDivider",
+            Gaffer.Color3fPlug,
+            Gaffer.Color4fPlug,
+        ] :
+            if isinstance( item, six.string_types ) :
+                result.append( "/" + item, { "divider": True } )
+            else :
+                result.append(
+                    "/" + item.__name__.replace( "Plug", "" ),
+                    {
+                        "command" : functools.partial( Gaffer.WeakMethod( self.__addQuery ), "parameter", item ),
```

Comment:
I'm not happy with the default value of `"parameter"` here, but I was having the following problem :

1. Pass an empty string instead of `"parameter"`
2. The output child plugs of `exists` and `values` are added correctly, and with no name, as expected.
3. If the children of `exists` or `values` are visible by expanding the dropdown when the query is added, they will never update when changing the query parameter name.
4. If the children of `exists` or `values` are _not_ visible when the query is added, the query name is changed, and the  children are made visible by expanding the dropdown, they update as expected when changing the query parameter.

Couldn't find anything indicating `label` visibility, so this is a workaround for now.

Reply:
I suppose I'll resolve my own conversation - it was fixed in 6da2cd62dd76b0ba050248530a7f41e9173f3e7a